earthmover/nodes/destination.py

```diff
@@ -43,53 +43,47 @@ class FileDestination(Destination):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.template = self.error_handler.assert_get_key(self.config, 'template', dtype=str)
-        self.header = self.config.get("header")
-        self.footer = self.config.get("footer")
+        self.header = self.error_handler.assert_get_key(self.config, 'header', dtype=str, required=False, default=None)
+        self.footer = self.error_handler.assert_get_key(self.config, 'footer', dtype=str, required=False, default=None)
+        self.linearize = self.error_handler.assert_get_key(self.config, 'linearize', dtype=bool, required=False, default=True)
+        self.extension = self.error_handler.assert_get_key(self.config, 'extension', dtype=str, required=False, default='')
 
         #config->extension is optional: if not present, we assume the destination name has an extension
-        extension = ""
-        if "extension" in self.config:
-            extension = f".{self.config['extension']}"
-            
-        self.file = os.path.join(
-            self.earthmover.state_configs['output_dir'],
-            f"{self.name}{extension}"
-        )
+        filename = f"{self.name}.{self.extension}" if self.extension else self.name
+        self.file = os.path.join(self.earthmover.state_configs['output_dir'], filename)
+
+    def execute(self, **kwargs):
+        """
+        There is a bug in Dask where `dd.to_csv(mode='a', single_file=True)` fails.
+        This is resolved in 2023.8.1: https://docs.dask.org/en/stable/changelog.html#id7 
+
+        :return:
+        """
```

Comment:
@jayckaiser I must be going crazy - the above comment (about the Dask bug) has been removed in the current `rc/0.3.2` branch (due to the refactor of how destination files are written); why isn't github showing a merge conflict on this PR (which is against `rc/0.3.2`)?

Reply:
So this will happen if both of the branches branched from `rc/0.3.2`, instead of `main`. The deletion is accounted for in the history, so there's no conflict.

That's my best understanding of it, anyway 😉 Git is an arcane beast.